### PR TITLE
[FEATURE] Introduce Twig for templating

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

File: Classes/Util/ClassDocsHelper.php [hunk 1]
```diff
@@ -414,7 +425,28 @@ public static function extractPhpDomain(
...
+            'classSignature' => $classReflection->getShortName(),
+            'modifiers' => $modifiers,
+            'classComment' => self::getClassComment($classReflection, $gitHubLink, $includeClassComment),
+            'noindexInClass' => $noindexInClass,
+            'classBody' => $classBody,
+            'type' => $type,
+        ];
+
+        // Render the template
+        return $twig->render('phpClass.rst.twig', $data);
```
Comment (froemken) [inline Classes/Util/ClassDocsHelper.php]:
```suggestion
        $variables = [
            'namespace' => $classReflection->getNamespaceName(),
            'classSignature' => $classReflection->getShortName(),
            'modifiers' => $modifiers,
            'classComment' => self::getClassComment($classReflection, $gitHubLink, $includeClassComment),
            'noindexInClass' => $noindexInClass,
            'classBody' => $classBody,
            'type' => $type,
        ];

        // Render the template
        return $twig->render('phpClass.rst.twig', $variables);
```

File: Resources/Private/Templates/phpClass.rst.twig [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{% if namespace -%}
```
Comment (froemken) [inline Resources/Private/Templates/phpClass.rst.twig:1]: What is `-%}`? Never saw that: https://twig.symfony.com/doc/3.x/tags/if.html

**Review by froemken — CHANGES_REQUESTED** [review]
Nice work. But I and `$data` don't get friends. Call it `$variables`, `$placeholders` or...hey...what about `$markers` ;-)

File: Resources/Private/Templates/phpClass.rst.twig (hunk 2 above)
Comment (linawolf) [reply Resources/Private/Templates/phpClass.rst.twig:1]: Those are whitespace controll options: https://symfony.com/blog/better-white-space-control-in-twig-templates

File: Classes/Util/ClassDocsHelper.php (hunk 1 above)
Comment (linawolf) [reply Classes/Util/ClassDocsHelper.php]: The variable is actually called $context in the render method of twig. so lets stick to that?

**Review by froemken — APPROVED** [review]

File: Classes/Util/ClassDocsHelper.php (hunk 1 above)
Comment (froemken) [reply Classes/Util/ClassDocsHelper.php]: Everything is better than `$data`. Approved

**Pull request merged**